### PR TITLE
feat: NuGet C# parser

### PR DESCRIPTION
**Pull request by gautambhat**
PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [ ] Documentation is up-to-date
- [ ] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: cli/tests/e2e/test_dependency_aware_rules.py [hunk 1]
```diff
@@ -154,6 +154,15 @@
...
         ),
+        (
+            "rules/dependency_aware/nuget-sca-simple.yaml",
+            "dependency_aware/nuget",
+        ),
+        # This test intentionally runs poetry rules against C# to check that scan runs correctly and does not produce findings
+        (
+            "rules/dependency_aware/python-poetry-sca.yaml",
+            "dependency_aware/nuget",
+        ),
```
Comment (gautambhat) [inline cli/tests/e2e/test_dependency_aware_rules.py]: Not adding more tests until I decide on what direction to take wrt same dependency instances across different target frameworks.

File: cli/src/semdep/parse_lockfile.py [hunk 2]
```diff
@@ -97,6 +101,8 @@ def parse_dep(s: str) -> FoundDependency:
     "maven_dep_tree.txt": None,
     "gradle.lockfile": "build.gradle",
     "pnpm-lock.yaml": None,
+    "packages.lock.json": None,
+    "packages.config": "packages.config",
```
Comment (mmcqd) [inline cli/src/semdep/parse_lockfile.py]: What's this?

File: cli/tests/e2e/test_dependency_aware_rules.py (hunk 1 above)
Comment (mmcqd) [reply cli/tests/e2e/test_dependency_aware_rules.py]: I believe you said this got resolved?

**Review by mmcqd — APPROVED** [review]
This looks good. I think it'd be worth trying to scrape some huge number of `packages.lock.json` files from github or something to see if we can parse them. In the past we've had issues with insufficiently tested parsers, and now is as good a time as any to improve the situation.

File: cli/tests/e2e/test_dependency_aware_rules.py (hunk 1 above)
Comment (gautambhat) [reply cli/tests/e2e/test_dependency_aware_rules.py]: There's a little bit of contention around how we want to solve this, so I'm setting up a project in Snyk to check its behaviour 🥼 🐀

File: cli/src/semdep/parse_lockfile.py (hunk 2 above)
Comment (gautambhat) [reply cli/src/semdep/parse_lockfile.py]: oops I added this temporarily, and forgot to remove it.

**Pull request merged**